### PR TITLE
levant: init at 0.3.0

### PR DESCRIPTION
**Pull request by max-niederman**
###### Motivation for this change

It would be useful to have [Levant](https://github.com/hashicorp/levant), a templating tool for [Nomad](https://www.nomadproject.io/) jobs available.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7202,6 +7202,16 @@
...
+  max-niederman = {
+    email = "max@maxniederman.com";
+    github = "max-niederman";
+    githubId = 19580458;
+    name = "Max Niederman";
+    keys = [{
+      longkeyid = "rsa3072/0x9AED881481D8444E";
+      fingerprint = "1DE4 424D BF77 1192 5DC4  CF5E 9AED 8814 81D8 444E";
+    }];
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:7214]: Separate commit, `maintainers: add max-niederman`

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 145548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>levant</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/145548#issuecomment-966945108)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-145548]$ ./results/levant/bin/levant 
Usage: levant [--version] [--help] <command> [<args>]

Available commands are:
    deploy       Render and deploy a Nomad job from a template
[...]                                                                                                              │
```

Looks good to me.

File: pkgs/applications/networking/cluster/levant/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  # The tests try to connect to a Nomad cluster.
+  doCheck = false;
+
+  meta = with lib; {
+    description = "An open source templating and deployment tool for HashiCorp Nomad jobs";
+    homepage = "https://github.com/hashicorp/levant";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ max-niederman ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/levant/default.nix:26]:
```suggestion
```
default of buildGoModule

Comment (max-niederman) [reply pkgs/applications/networking/cluster/levant/default.nix:26]: It's actually not. `buildGoModule` [inherits from](https://github.com/NixOS/nixpkgs/blob/2d47affbfb2d961b721a435513710ee7921d3dfd/pkgs/development/go-modules/generic/default.nix#L267) [Go](https://github.com/NixOS/nixpkgs/blob/2d47affbfb2d961b721a435513710ee7921d3dfd/pkgs/development/compilers/go/1.17.nix#L275), which is `platforms.linux ++ platforms.darwin`. I was able to verify this by checking the platforms with `nix-env -qa` with and without the platforms line. Admittedly, I have no idea whether the package would build on Cygwin, other BSDs, or Redox, but that's the platform [Nomad's package uses](https://github.com/NixOS/nixpkgs/blob/2d47affbfb2d961b721a435513710ee7921d3dfd/pkgs/applications/networking/cluster/nomad/generic.nix#L45).

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 145548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>levant</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/levant/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/applications/networking/cluster/levant/default.nix:26]: I think that's close enough. If it causes issues, we can revisit this.

**Review by jonringer — APPROVED** [review]
LGTM
shows usage

```
https://github.com/NixOS/nixpkgs/pull/145548

1 package built:
levant
```

File: pkgs/applications/networking/cluster/levant/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/cluster/levant/default.nix:26]: buildGoModule should not set platforms.unix or Linux + Darwin. The same is true for equivalent python and rust function. If those package sets ever support another platform we don't want to update hundreds of packages which just work on the new platform.

Comment (max-niederman) [reply pkgs/applications/networking/cluster/levant/default.nix:26]: I'm not totally sure what you mean. Are you saying that _consumers_ of `buildGoModule` shouldn't set `meta.platforms` or that `buildGoModule` itself shouldn't?

**Pull request merged**